windows/gce-gpu-monitoring-cuda.ps1

```diff
@@ -0,0 +1,227 @@
+                    }
+                'value'=@{
+                    'int64Value'= $value
+                    }
+                    
+            }        
+        )
+    }
+}
+
+function check_nvidia_smi{
+    <#
+    Checks if the nvidia-smi tool is installed and if it detects any GPUs.
+    Prints message to stderr in case of errors.
+    =return=
+    True if the nvidia-smi tool is available.
+    #>
+
+    $installed = test-path -Path $nvidia_smi_path
+    if (-not($installed)){
+        write-host -ForegroundColor Red "Couldn't find the nvidia-smi tool. Make sure it's properly installed in one of the directories in $nvidia_smi_path."
+        return $false
+    }
+
+    $result = & $nvidia_smi_path -L 
+    if ($LASTEXITCODE -ne 0){
+        write-host -ForegroundColor Red "The nvidia-smi tool has encountered an error= "
+        write-host -ForegroundColor Red "$result"
+        return $false
+    }
+
+    <# 
+    $no_cards = $result|Measure-Object -Line
+    if ( $no_cards -eq 0 ){
+        Write-host -ForegroundColor RED "The nvidia-smi tool didn't detect any GPUs attached to the system."
+        return $false
+    }
+    #>
+    
+    return $true
+}
+
+function report_metrics{
+    <#
+    Reports a set of metrics to the Cloud Monitoring system.
+    :param values: A dictionary mapping (gpu_type, gpu_bus_id) to a map of
+        metric names and their values (floats).
+    :param project_id: Project number of the hosting machine.
+    :param zone: Zone of the hosting machine.
+    :param instance_id: Instance ID of the hosting machine.
+    :return:
+    #>
+    param(
+        $metrics
+    )
+
+    $data = @{'timeSeries'=@()}
+
+    #gpu total
+    $now=(get-date).ToString("O")
+
+    # $gpu = get_nvidia_smi_utilization -metric_name "utilization.gpu"
+    # $memory = get_nvidia_smi_utilization -metric_name "utilization.memory"
+    
+    foreach ($key in $metrics.keys){
+        $gpu = $key -split ","
+
+        foreach($it in $metrics[$key].keys){
+            $data.timeSeries += get_timeseries_entry -metric_time $now -nvidia_metric $it -gcp_metric_name $gpu_metrics[$it] -value $metrics[$key][$it] -gpu_type $gpu[0] -gpu_bus_id $gpu[1]    
+        }
+    }
+
+
+
+    
+    # $data.timeSeries += get_timeseries_entry -metric_time $now -nvidia_metric "utilization.memory" -gcp_metric_name "gpu_memory_utilization"
+
+
+    $body = $data | ConvertTo-Json -Depth 6
+
+    $body
+    try{
+        $result=Invoke-RestMethod -Method Post -Headers $headers -Uri "https=//monitoring.googleapis.com/v3/projects/$project_id/timeSeries" -Body $body    
+    }
+    catch{
+        #if the token is expired, set it again and send the metrics
+        $access_token = Invoke-Expression $access_token_command
+        $headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+        $headers.Add("Authorization", "Bearer "+$access_token)
+        $headers.Add("Content-Type", 'application/json; charset=utf-8') 
```

Comment:
These lines may be simplified as following:
```
$headers = @{
  Authorization = "Bearer $access_token"
}
```

And move the `Content-Type` to using the [`Invoke-RestMethod` `-ContentType` parameter](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-5.1).

Reply:
updated as advised. Thanks.